src/internal_events/mod.rs

```diff
@@ -0,0 +1,40 @@
+mod blackhole;
+mod elasticsearch;
+mod file;
+mod lua;
+mod prometheus;
+mod regex;
+mod syslog;
+mod tcp;
+mod udp;
+mod unix;
+mod vector;
+
+pub use self::blackhole::*;
+pub use self::elasticsearch::*;
+pub use self::file::*;
+pub use self::lua::*;
+pub use self::prometheus::*;
+pub use self::regex::*;
+pub use self::syslog::*;
+pub use self::tcp::*;
+pub use self::udp::*;
+pub use self::unix::*;
+pub use self::vector::*;
+
+pub trait InternalEvent: Sized {
```

Comment:
I started with something like that, but then it gets confusing because you can't really communicate that people should only override `emit` xor `emit_logs`/`emit_metrics`. It seemed simpler to just give the two that don't conflict.